### ISSUE TITLE
`azurerm_cosmosdb_account` fails with `Error: missing expected [` when updating from `azurerm` provider `v3.116.0` to `v4.0.1`

### ISSUE THREAD

**Issue by chamindac** [hashicorp/terraform-provider-azurerm#27242]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9.5

### AzureRM Provider Version

4.0.1

### Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_account

### Terraform Configuration Files

```hcl
terraform {
  backend "local" {
    path = "terraform.tfstate"
  }
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      # version = "=3.116.0"
      version = "=4.0.1"
    }
  }
}

provider "azurerm" {
  features {}
  subscription_id = "subscriptionid"
}

resource "azurerm_resource_group" "instance_rg" {
  name     = "ch-cosmostest-dev-wus-001-rg"
  location = "westus"
}

resource "azurerm_cosmosdb_account" "instancecosmos" {
  name                          = "ch-cosmostest-dev-wus-001-cdb"
  location                      = azurerm_resource_group.instance_rg.location
  resource_group_name           = azurerm_resource_group.instance_rg.name
  offer_type                    = "Standard"
  kind                          = "GlobalDocumentDB"
  automatic_failover_enabled    = false
  public_network_access_enabled = true
  default_identity_type = "FirstPartyIdentity"

  geo_location {
    location          = azurerm_resource_group.instance_rg.location
    failover_priority = 0
  }

  consistency_policy {
    consistency_level       = "Session"
    max_interval_in_seconds = 5
    max_staleness_prefix    = 100
  }
}
```

### Debug Output/Panic Output

```shell
azurerm_resource_group.instance_rg: Refreshing state... [id=/subscriptions/subscription/resourceGroups/ch-cosmostest-dev-wus-001-rg]

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: missing expected [
│
│   with azurerm_cosmosdb_account.instancecosmos,
│   on main.tf line 24, in resource "azurerm_cosmosdb_account" "instancecosmos":
│   24: resource "azurerm_cosmosdb_account" "instancecosmos" {
```

### Expected Behaviour

Plan should execute with no change to cosmos db account

### Actual Behaviour

![image](https://github.com/user-attachments/assets/49d2f631-e16f-479f-a88d-4e891ae3417b)

### Steps to Reproduce

Follwing below steps causes the issue mentioned.

1. Use the provided terraform configuration file and create a resource group and cosmos DB account with `azurerm` provider set as `version = "=3.116.0" `. 
2. Run `terraform init`, `terraform plan -out='my.tfplan'` and `terraform apply my.tfplan`
3. Resource group and cosmos db account gets created. 
4. Update `azurerm` to `version = "=4.0.1"` 
5. Run `terraform init -upgrade`  and then  `terraform plan -out='my.tfplan'`
![image](https://github.com/user-attachments/assets/c532ef6f-473c-4cdc-9171-1de2e3e986ff)
 
 
 However, with the same terraform consfiguration files above using `azurerm` with `version = "=4.0.1"`, creating a fresh resource group and  a cosmos db account works fine.
 
 In summary,

- Above terraform configuration files can be used with `azurerm` provider  `version = "=4.0.1"` or `version = "=3.116.0" ` to create a fresh cosmos db account without any issue.
- If the cosmos db account is created with `azurerm` provider `version = "=3.116.0" ` and then updating  `azurerm` provider to  `version = "=4.0.1"` causes  `Error: missing expected [` in `terraform plan`
 

### Important Factoids

I am using Azure public cloud

### References

N/A

**Comment by raswinraaj** [comment]
We are unable to upgrade to azurerm 4.0 until this issue is resolved .

**Comment by raswinraaj** [comment]
I don't have ip_range_filter in my code.

**Comment by chamindac** [comment]
@catriona-m  Without IP range filter as well this issue can be reproduced as in my attached code sample

**Comment by roschart** [comment]
We are encountering the same issue with updating the AzureRM provider to version 4.0.1. In our case, we are also not using `ip_range_filter` in the code.

We attempted to delete the Terraform state and re-import it, but we ran into difficulties due to dynamically generated resources, which further complicated the process. Ultimately, we decided to abort the update and roll back the code, which resolved the issue.

As a result, we are now waiting for this issue to be resolved before updating the AzureRM provider.

**Comment by svaraksin-gd** [comment]
Agree. We also found it difficult to update dozens of state files. 
Are there any plans to make it easily upgradable in the next release? @stephybun

**Comment by dagl** [comment]
Updating the line manually in tfstate
`"ip_range_filter": "",`
to 
`"ip_range_filter": [""],`
also fixes the issue

**Comment by orgads** [comment]
> Updating this line manually in tfstate also fixes the issue: `"ip_range_filter": "",` to `"ip_range_filter": [""],`

Actually it should be `[]`, not `[""]`.

**Comment by raswinraaj** [comment]
@jackofallops can you please let us know the version in which this issue is resolved?

**Comment by WolfyUK** [comment]
This also seems to affect `partition_key_path`/`s` on [cosmosdb_sql_container](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cosmosdb_sql_container)

**Comment by chamindac** [comment]
@raswinraaj I tested with latest [azurerm 4.1.0  ](https://github.com/hashicorp/terraform-provider-azurerm/releases). The issue is now fixed. Thanks @jackofallops

**Comment by orgads** [comment]
I already applied on 4.0.1 with the hack above, and now when I try to apply with 4.1.0 it crashes.

- #27308
